inc/tickets.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace WPBerry;
+
+class Tickets {
+
+	public function __construct() {
+		add_action( 'save_post', array( $this, 'modify_slug_on_save' ), 10, 3 );
+	}
+
+
+	public function modify_slug_on_save( $post_id, $post, $update ) {
+		// allow 'publish', 'draft', 'future'.
+		if ( $post->post_type != 'berry' || $post->post_status == 'auto-draft' ) {
+			return;
+		}
+		// only change slug when the post is created (both dates are equal).
+		if ( $post->post_date_gmt != $post->post_modified_gmt ) {
+			return;
+		}
+
+		$new_slug = $this->get_ticket_slug( $post_id );
+		if ( $new_slug == $post->post_name ) {
+			return;
+		}
+
+		// unhook this function to prevent infinite looping.
+		remove_action( 'save_post', 'slug_save_post_callback', 10, 3 );
+		// update the post slug (WP handles unique post slug)
+		wp_update_post(
+			array(
+				'ID'        => $post_id,
+				'post_name' => $new_slug,
+			)
+		);
+		// re-hook this function.
+		add_action( 'save_post', 'slug_save_post_callback', 10, 3 );
+	}
+
+	public function get_ticket_slug( $post_id ) {
```

Comment:
We can modify the function name to something different indicating it will return a new ticket slug. Maybe `get_new_slug_for_ticket`